tests/Laravel/Metadata/VisibleHiddenAttributesTest.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Tests\Laravel\Metadata;
+
+use ApiPlatform\Laravel\Eloquent\Metadata\ModelMetadata;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @author Tobias Oitzinger <tobiasoitzinger@gmail.com>
+ */
+class VisibleHiddenAttributesTest extends TestCase
+{
+    private ModelMetadata $modelMetadata;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->modelMetadata = new ModelMetadata();
+    }
+
+    protected function tearDown(): void
+    {
+        \Mockery::close();
+        parent::tearDown();
+    }
+
+    public function testHiddenAttributesAreCorrectlyIdentified(): void
+    {
+        $model = new class extends Model {
+            protected $hidden = ['secret'];
+        };
+
+        $result = $this->invokePrivateMethod($this->modelMetadata, 'attributeIsHidden', ['secret', $model]);
+        $this->assertTrue($result);
+
+        $result = $this->invokePrivateMethod($this->modelMetadata, 'attributeIsHidden', ['public', $model]);
+        $this->assertFalse($result);
+    }
+
+    public function testVisibleAttributesAreCorrectlyIdentified(): void
+    {
+        $model = new class extends Model {
+            protected $visible = ['public'];
+        };
+
+        $result = $this->invokePrivateMethod($this->modelMetadata, 'attributeIsHidden', ['secret', $model]);
+        $this->assertTrue($result);
+
+        $result = $this->invokePrivateMethod($this->modelMetadata, 'attributeIsHidden', ['public', $model]);
+        $this->assertFalse($result);
+    }
+
+    public function testAllAttributesVisibleByDefault(): void
+    {
+        $model = new class extends Model {};
+
+        $result = $this->invokePrivateMethod($this->modelMetadata, 'attributeIsHidden', ['any_attribute', $model]);
+        $this->assertFalse($result);
+    }
+
+    public function testGetRelationsReturnsCorrectRelations(): void
+    {
+        $relation = \Mockery::mock(HasMany::class);
```

Comment:
We don't use Mockery on our code base, maybe that a functional test would've been enough for this use case? If you want to unit test this, you may test the result of `getRelations`.